source/Calamari.Common/Features/Discovery/AccountAuthenticationDetails.cs

```diff
@@ -17,14 +16,6 @@ public AccountAuthenticationDetails
             AccountDetails = accountDetails;
         }
 
-        public AccountAuthenticationDetails(
```

Comment:
Overall I don't have a good understanding of target discovery and this change is more of a spike/showing what I think this should look like. Please let me know if there are any issues with this change.

Deserializing with both constructors was throwing errors:
`Deserialization of types without a parameterless constructor, a singular parameterized constructor, or a parameterized constructor annotated with 'JsonConstructorAttribute' is not supported. Type 'Calamari.Common.Features.Discovery.AccountAuthenticationDetails`1[Calamari.AzureAppService.Azure.ServicePrincipalAccount]'. Path: $.authentication | LineNumber: 8 | BytePositionInLine: 23.`

You cannot have two JsonConstructor attributes to a class and I'm assuming the original constructor with a null type was maintained for Calamari Flavours that aren't updated to include the Type property. 
I've added this property in the test config and am assuming that targetDiscoveryContext coming in from server will have a Type attribute for the AzureAppService, is this the case?

Reply:
It's weird that it threw that error as the top constructor is marked with `[JsonConstructor]`... Either way I think you're right about why the original constructor still exists. Are all those flavours merged into this solution now?

Reply:
Yeah I was surprised by this behaviour, it really didn't make sense to me but this way works... Currently we only have AzureAppService, AzureResourceGroup and soon to be AzureWebApp migrated. All other packages would be using a pinned package reference for Calamari.Common so even if this deprecated support for the standalone flavour projects I don't think that's an issue. We'll be updating to project references when migrating and having to make sure tests pass/this doesn't cause an issue anyway.

Reply:
I think the only other flavour that would use this type would be AzureWebApp as it's the only other step which has target discovery. (AWS ECS also supported target discovery but it's in a step package which means it's done differently).
As long as AzureWebApp is happy I think this change should be ok.

Reply:
I believe kubernetes discovery will also use this class for both AKS and EKS, so might be worth checking that too.

Reply:
Hrm yes you're right. I'm happy to run through doing discovery with you @IsaacCalligeros95 on AzureWebApp, K8s EKS/AKS to smoke test it, if you want. I'll try and get my PR to Calamar in first so that AKS is fixed.

Reply:
@gb-8 might also remember if there were any reasons this extra constructor was included previously. One thought is that older versions of Server may not have sent the `Type` property as part of the authentication property on the discovery context, assuming these changes are only going to the latest Server release and not back to LTS branches then removing the constructor I imagine would be fine.

Reply:
@scottmerchant A walk-through target discovery would be good, just let me know when it suits you